src/sestest/VerifyCommand.cs

```diff
@@ -0,0 +1,164 @@
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web;
+using Microsoft.Azure.Batch.SoftwareEntitlement.Common;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.Batch.SoftwareEntitlement
+{
+    /// <summary>
+    /// Functionality for the <c>submit</c> command
+    /// </summary>
+    public sealed class VerifyCommand : CommandBase
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="VerifyCommand"/> class
+        /// </summary>
+        /// <param name="logger">A logger to use while executing.</param>
+        public VerifyCommand(ILogger logger) : base(logger)
+        {
+        }
+
+        /// <summary>
+        /// Generate a new token
+        /// </summary>
+        /// <param name="commandLine">Configuration from the command line.</param>
+        /// <returns>Results of execution (0 = success).</returns>
+        public async Task<int> Execute(VerifyCommandLine commandLine)
+        {
+            Errorable<Uri> server = FindServer(commandLine);
+            Errorable<string> token = FindToken(commandLine);
+            Errorable<string> app = FindApplication(commandLine);
+            Errorable<string> api = FindApiVersion(commandLine);
+
+            Errorable<string> result = await server.With(token).With(app).With(api)
+                .MapAsync(SubmitToken);
+            return result.Match(response =>
+                {
+                    Logger.LogJson(response);
+                    return ResultCodes.Success;
+                },
+                errors =>
+                {
+                    Logger.LogErrors(errors);
+                    return ResultCodes.Failed;
+                });
+        }
+
+        private async Task<string> SubmitToken(Uri server, string token, string app, string api)
+        {
+            var builder = new UriBuilder(server);
+            //TODO: Do this in a path safe way similar to Path.Combine
+            builder.Path = builder.Path + "softwareEntitlements";
+            builder.Query = CreateParameters(("api-version", api));
+
+            var uri = builder.Uri;
+
+            using (var client = new HttpClient())
```

Comment:
For best design it might be better to inject the `HttpClient` to allow it to be shared.

Reply:
This is a single shot method that runs, then the process exits; injection would just add more complexity but wouldn't benefit performance. #wontfix